Watchman.Discord/Areas/Administration/Controllers/MutedUsersController.cs

```diff
@@ -30,27 +29,36 @@ public MutedUsersController(DirectMessagesService directMessagesService, MutingH
         [AdminCommand]
         public async Task MutedUsers(MutedUsersCommand mutedUsersCommand, Contexts contexts)
         {
-            var mutedUsersList = await _usersService.GetUsersAsync(contexts.Server).ToListAsync();
-            var (title, description, values) = this.GetMuteEmbedMessage(mutedUsersList, contexts.Server.Id);
-            await _directMessagesService.TrySendEmbedMessage(contexts.User.Id, title, description, values);
+            var mutedUsers = _usersService.GetUsersAsync(contexts.Server);
+            var (title, description, values) = await this.GetMuteEmbedMessage(mutedUsers, contexts.Server.Id);
+            if (values.Count == 0)
+            {
+                await _directMessagesService.TrySendMessage(contexts.User.Id, "Brak wyciszonych użytkowników!");
+            }
+            else
+            {
+                await _directMessagesService.TrySendEmbedMessage(contexts.User.Id, title, description, values);
+            }
         }
-        private (string title, string description, Dictionary<string, Dictionary<string, string>> values) GetMuteEmbedMessage(List<UserContext> mutedUsersList, ulong serverId)
+
+        private async Task<(string title, string description, Dictionary<string, Dictionary<string, string>> values)> GetMuteEmbedMessage(IAsyncEnumerable<UserContext> mutedUsers, ulong serverId)
         {
             var title = "Lista wyciszonych użytkowników";
             var description = "Wyciszeni użytkownicy, powody oraz data wygaśnięcia";
             var values = new Dictionary<string, Dictionary<string, string>>();
-            foreach (var mutedUser in mutedUsersList)
+            await foreach (var mutedUser in mutedUsers)
             {
                 var muteEvent = this._mutingHelper.GetNotUnmutedUserMuteEvent(serverId, mutedUser.Id);
-                if (muteEvent != null)
+                if (muteEvent == null || DateTime.Compare(muteEvent.TimeRange.End.ToLocalTime(), DateTime.Now) < 0)
```

Comment:
taka sytuacja nie powinna się wydarzyć, że czas już minął, ponieważ GetNotUnmutedUserMuteEvent zwraca tylko wciąż wyciszony muteEvent

Reply:
dopisałam ten warunek właśnie dlatego, że GetNotUnmutedUserMuteEvent zwrócił mi już nie wyciszony muteEvent. tu przykład bez warunku:
![image](https://user-images.githubusercontent.com/35144936/89311556-72479400-d676-11ea-9614-f70e99fe36fa.png)

Reply:
To oznacza poważny błąd
Sprawdź dlaczego tak się dzieje u Ciebie, ponieważ u mnie to działa dobrze i nigdy nie zwraca unmuted users
Zobacz czy nie są unmutowani czy źle są filtrowane muteEvents z bazy

Reply:
zauważyłam, że u mnie się to dzieje tylko wtedy, gdy wyłączę bota zanim użytkownik zostanie odmutowany, w innym przypadku jest wszystko ok

Reply:
i po ręcznym odmutowaniu użytkownika też już jest wszystko ok

Reply:
aa, bo uruchamiasz Watchman.Discord, a nie Watchman.Web?
jeśli tak, to po prostu tam nie ma uruchamiane go Hangfire'a który tym się zajmuje

Reply:
na produkcji zawsze jest wersja webowa, dlatego nie musisz się przejmować tym problemem
(jeśli masz czas i chęci to możesz potestować w dockerze tak jak pisałem, wtedy powinno to działać)

Reply:
ok, dzięki za informację :)